src/test/java/org/opensearch/flowframework/transport/CreateWorkflowTransportActionTests.java

```diff
@@ -132,7 +144,16 @@ public void setUp() throws Exception {
         );
     }
 
-    public void testFailedDryRunValidation() {
+    public void testDryRunValidation_Success() {
+        Template validTemplate = generateValidTemplate();
+
+        @SuppressWarnings("unchecked")
+        ActionListener<WorkflowResponse> listener = mock(ActionListener.class);
+        WorkflowRequest createNewWorkflow = new WorkflowRequest(null, validTemplate, true, null, null);
+        createWorkflowTransportAction.doExecute(mock(Task.class), createNewWorkflow, listener);
```

Comment:
You should be able to capture the listener argument and verify onResponse was called

Reply:
This test case primarily tests the logic in the validateWorkflows method, which occurs before any ActionListener is involved.